### PR TITLE
restructured service interfaces to avoid duplicate code

### PR DESCRIPTION
**Pull request by thomas-ayling**
Restructuring services etc

### CODE REVIEW

**Review by MihneaDinuGL — APPROVED** [review]
Nice work!

File: src/main/java/com/globallogic/amcr/service/Service.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.globallogic.amcr.service;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface Service<T> {
```
[the remark this answers is not in the document]
Comment (thomas-ayling) [reply src/main/java/com/globallogic/amcr/service/Service.java:6]: Oh I see... no problem I will revert

File: src/main/java/com/globallogic/amcr/service/casestudies/CaseStudyService.java [hunk 2]
```diff
@@ -1,70 +1,33 @@
...
-     */
-    CaseStudy get(UUID id);
-
-    /**
-     * requests all case studies in the database
-     *
-     * @return returns a list of all saved case studies with all fields
-     */
-    List<CaseStudy> getAll();
+public interface CaseStudyService extends CrudService<CaseStudy> {
```
[the remark this answers is not in the document]
Comment (thomas-ayling) [reply src/main/java/com/globallogic/amcr/service/casestudies/CaseStudyService.java:9]: Will do

**Comment by thomas-ayling** [comment]
Reverted those changes to the services @sourcesmith do you mind accepting if everything else is acceptable?

**Pull request merged**